sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcSourceSuite.scala

```diff
@@ -581,7 +580,7 @@ class OrcSourceSuite extends OrcSuite with SharedSparkSession {
   }
 
   test("Enforce direct encoding column-wise selectively") {
-    testSelectiveDictionaryEncoding(isSelective = true)
+    testSelectiveDictionaryEncoding(isSelective = true, isHive23 = false)
```

Comment:
why `isHive23` is false here?

Reply:
This is `sql/core` module. We are using Apache ORC directly.

Reply:
This test case is added originally via [[SPARK-25635][SQL][BUILD] Support selective direct encoding in native ORC write](https://github.com/apache/spark/commit/1c9486c1acceb73e5cc6f1fa684b6d992e187a9a) .

Reply:
After that, [[SPARK-27402][SQL][TEST-HADOOP3.2][TEST-MAVEN] Fix hadoop-3.2 test issue(except the hive-thriftserver module](https://github.com/apache/spark/commit/f3ddd6f9da27925607c06e55cdfb9a809633238b#diff-c0ffc4377a037f2e6def9779d0ec2527R121
) changed like the following.

```scala
- protected def testSelectiveDictionaryEncoding(isSelective: Boolean) {
+ protected def testSelectiveDictionaryEncoding(isSelective: Boolean, isHive23: Boolean = false) {
```

Reply:
Initially, SPARK-27402 seems to use `isHive23` misleadingly because `isHive23=false` means Apache ORC in `sql/core` module and Hive 1.2.1 fork in `sql/hive` module.

Reply:
cc @wangyum and @gatorsmile

Reply:
shall we rename to `isHiveOrc`?

Reply:
Sure!